02-oop-inheritance/javaScript/refactored_JS/downloableMovie v0.1.js

```diff
@@ -0,0 +1,94 @@
+var MovieObserver = function ( ) {};
+MovieObserver.prototype.playing = function ( title ) {
+    console.log('A movie with title ' + title + ' has been played');
+};
+MovieObserver.prototype.stopped = function( title ) {
+    console.log('A movie with title ' + title + ' stopped');
+};
+
+var Movie = function ( paramName, paramValue ) {
+  //Variables
+  this.attributes = [];
+  this.attributes[paramName] = paramValue; 
+  this.observer = new MovieObserver();
+};//end movie constructor
+
+Movie.prototype = {
+  
+  get: function (atribute) {
+     return this.attributes[atribute];
```

Comment:
This works but is not quite right semantically. An array usually is used with numeric indexes not with key indexes. That's why the comment. But this is related with other comment. Anyway if you try to do this

this.attributes['sarasa'] = 'a';
console.log(this.attributes.sarasa)

this will console undefined. Have that in mind :/